### DISCUSSION TITLE
local-optimization MINLP solver compatible with YALMIP

### DISCUSSION THREAD

**Discussion by asxadf** [yalmip/YALMIP#1080] [Optimization modelling] [answered]
Hi Dr. Johan,

I am looking for an MINLP solver compatible with YALMIP, such as CONOPT (But not with YALMIP). Instead of global optimality, I prefer to get better computation speed. (Currently, I am working with the complementary conditions in KKT. I want to try some quick local optimization instead of Big-M.)

I noticed that you have a list of "General nonlinear programming and other solvers" at https://yalmip.github.io/allsolvers/. However, only BARON remains state-of-the-art, and the others stop their development somehow. Additionally, BARON pursues global optimum. Thus, I am concerned about the speed.

Could you please recommend any MINLP solver that can quickly provide local optimal solutions?

Best regards,
Xianbang

**Comment by johanlofberg** [comment, marked as answer]
knitro is local I think, so is bonmin and bnb (in the sense that they only branch in the integer space and thus assume the relaxations are solved to optimality)

**Reply by johanlofberg** [reply]
not sure why you talk about conopt though, as it is an NLP solver

**Reply by asxadf** [reply]
Because one paper I just read recommends CONOPT. I am not sure whether it can solve MINLP.

**Reply by asxadf** [reply]
Thank you, I am digging the knitro's website.

**Reply by asxadf** [reply]
Hi, Dr. Johan, I am reading the documents of KNITRO. Seems they have some internal function for complementary constraints. If I use YALMIP to call KNITRO for solving an MINLP with complementary constraints, whether YALMIP will activate the internal function for complementary constraints?

**Reply by asxadf** [reply]
I noticed that YALMIP internally supports SOS2, how about SOS1? I didn't find information about SOS1 on the YALMIP homepage.

**Reply by johanlofberg** [reply]
Any sensible integer solver will automatically detect sos1 structures from the constraints, if it has code exploiting this

**Reply by asxadf** [reply]
Thank you so much!
